cmd/credentialUtil.go

```diff
@@ -294,64 +405,40 @@ func getCredentialInfoForLocation(ctx context.Context, location common.Location,
 		} else {
 			credInfo.OAuthTokenInfo = *tokenInfo
 		}
+	} else if credInfo.CredentialType == common.ECredentialType.S3AccessKey() {
+		// TODO didn't seem to be any code for this in the original getCredentialInfoForLocation
+		//   but presumably we need some, based only on the fact that there are fields in credInfo for S3 stuff ,but maybe
+		//   those fields are never used?
 	}
```

Comment:
S3 auth is a weird, complicated beast. It happens when we get our S3 client, unlike the rest of our auth.

It'd be cool if we could refactor this in line with the rest of our auth, but it's not entirely critical.

Reply:
Ah, I see. Thanks. I'll change the TODO into a comment explaining that.